code/admin/forms/EventsForm.php

```diff
@@ -68,8 +68,14 @@ public function __construct($controller, $name)
             'Root.Past', _t('Event.PAST_EVENT_PLURAL','Past events')
         );
 
+        // Find all past events, including those with null start time
+        $time = date('Y-m-d', time());
+        $pastEvents = PublicEvent::get()
```

Comment:
@ToddHossack wouldn't it make more sense to make these changes directly to `CalendarHelper::past_events()`

Reply:
Yes, I initially revised CalendarHelper::past_events(), but then started thinking about it a bit more. The extra events the query would gather are basically invalid, and are only "past events" because their EndDateTime gets set to 1970... So if developers are using the past_events() method on their site, some invalid entries may suddenly appear in the front end after this change. The real purpose of the revision is only to allow management of those invalid events, rather than find more past events. Make sense? 
I know it's a bit DRYer in the past_events() method. If you think it's not a problem having it in the past_events method, I can move the change there. 

Cheers

Reply:
makes totally sense - thanks for clarifying!